### DISCUSSION TITLE
I am facing error on server

### DISCUSSION THREAD

**Discussion by Abdullahbutt3434** [spatie/browsershot#615] [Q&A]
I am implementing pdf download functionality of browsershot. It was working fine on my local but when I upload it on server it is showing me error kindly guide how I can resolve them all

error
Symfony\Component\Process\Exception\ProcessFailedException
The command "PATH=$PATH:/usr/local/bin NODE_PATH=`npm root -g` node '/home/staging/public_html/vendor/spatie/browsershot/src/../bin/browser.js' '{"url":"file:\/\/\/tmp\/1782316692-0224693001647935241\/index.html","action":"pdf","options":{"path":"\/home\/staging\/public_html\/storage\/file.pdf","args":[],"viewport":{"width":800,"height":600},"displayHeaderFooter":false,"margin":{"top":"60px","right":"50px","bottom":"1px","left":"40px"}}}'" failed. Exit Code: 1(General error) Working directory: /home/staging/public_html/en Output: ================ Error Output: ================ Error: Failed to launch the browser process! /home/staging/public_html/node_modules/puppeteer/.local-chromium/linux-970485/chrome-linux/chrome: error while loading shared libraries: libatk-1.0.so.0: cannot open shared object file: No such file or directory TROUBLESHOOTING: https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md at onClose (/home/staging/public_html/node_modules/puppeteer/lib/cjs/puppeteer/node/BrowserRunner.js:237:20) at Interface.<anonymous> (/home/staging/public_html/node_modules/puppeteer/lib/cjs/puppeteer/node/BrowserRunner.js:227:68) at Interface.emit (events.js:412:35) at Interface.close (readline.js:530:8) at Socket.onend (readline.js:254:10) at Socket.emit (events.js:412:35) at endReadableNT (internal/streams/readable.js:1334:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)

 my code:
               $invoice = Browsershot::html(view('wallet.orderInvoiceDownloadArticlePurchase')
                ->with('_orderDetails', $_orderDetails)->render())
                ->margins('60', '50', '1', '40', 'px')->save(storage_path('file.pdf'));
                
		$file= storage_path('file.pdf');
                $headers = array(
                     'Content-Type: application/pdf',
                );
                 return Response::download($file, 'order_'.$_orderDetails['orderId'].'Invoice.pdf', $headers)->deleteFileAfterSend(true);

**Comment by Lednerb** [comment]
```
 error while loading shared libraries: libatk-1.0.so.0: cannot open shared object file: No such file or directory
```

It seems that you have to install the needed dependencies on your server.
Maybe this one will help:
https://spatie.be/docs/browsershot/v2/requirements#installing-puppeteer-a-forge-provisioned-server

**Comment by singhvibhuti2** [comment]
I am exporting the PDF using below code and facing error.


        Browsershot::url('http://127.0.0.1:8000/ahip')->format('A2')->save('NationalReport1.pdf');
       
        $file = public_path('NationalReport1.pdf');
        $headers = ['Content-Type: application/pdf'];
        $updateNewPdfName = 'NationalReport'.time().'.pdf';


        return response()->download($file, $updateNewPdfName, $headers);

-> This view has a normal HTML and some amchart5 pie/donut charts. and throwing the below error. could you please help me to resolve this 


Symfony\Component\Process\Exception\ProcessFailedException
The command "node ^"C:^\Laravel Projects^\morningconsult^\vendor^\spatie^\browsershot^\src/../bin/browser.js^" ^"^{^\^"url^\^":^\^"http:^\/^\/127.0.0.1:8000^\/ahip^\^",^\^"action^\^":^\^"pdf^\^",^\^"options^\^":^{^\^"path^\^":^\^"NationalReport1.pdf^\^",^\^"args^\^":^[^],^\^"viewport^\^":^{^\^"width^\^":800,^\^"height^\^":600^},^\^"format^\^":^\^"A2^\^"^}^}^"" failed. Exit Code: 1(General error) Working directory: C:\Laravel Projects\morningconsult\public Output: ================ Error Output: ================ [1m[43m[30m Puppeteer old Headless deprecation warning:[0m[33m In the near future `headless: true` will default to the new Headless mode for Chrome instead of the old Headless implementation. For more information, please see https://developer.chrome.com/articles/new-headless/. Consider opting in early by passing `headless: "new"` to `puppeteer.launch()` If you encounter any bugs, please report them to https://github.com/puppeteer/puppeteer/issues/new/choose.[0m TimeoutError: Navigation timeout of 30000 ms exceeded at new Deferred (C:\Laravel Projects\morningconsult\node_modules\puppeteer-core\lib\cjs\puppeteer\util\Deferred.js:54:34) at Deferred.create (C:\Laravel Projects\morningconsult\node_modules\puppeteer-core\lib\cjs\puppeteer\util\Deferred.js:16:16) at new LifecycleWatcher (C:\Laravel Projects\morningconsult\node_modules\puppeteer-core\lib\cjs\puppeteer\cdp\LifecycleWatcher.js:65:60) at CdpFrame.goto (C:\Laravel Projects\morningconsult\node_modules\puppeteer-core\lib\cjs\puppeteer\cdp\Frame.js:135:29) at CdpFrame.<anonymous> (C:\Laravel Projects\morningconsult\node_modules\puppeteer-core\lib\cjs\puppeteer\util\decorators.js:98:27) at CdpPage.goto (C:\Laravel Projects\morningconsult\node_modules\puppeteer-core\lib\cjs\puppeteer\api\Page.js:603:43) at callChrome (C:\Laravel Projects\morningconsult\vendor\spatie\browsershot\bin\browser.js:171:20) at process.processTicksAndRejections (node:internal/process/task_queues:95:5) ERROR: The process "11732
